<?xml version="1.0" encoding="utf-8"?>
<launch>

  <arg name="model" default="quad" />

  <param name="/robot_description_semantic" textfile="$(find gap_passing)/config/moveit/hydrusx.srdf" />

  <node name="motion_planning_node" pkg="gap_passing" type="motion_planning_node" respawn="false" output="screen">
    <param name="coefficient_rate" value="0.05"/>
    <param name="planning_mode" value="1"/>
    <param name="motion_sequence_rate" value="50"/>
    <param name="ompl_mode" value="0"/> <!-- RRT START:0 -->

    <param name="gap_x_offset" value="0.6"/> <!-- 0.5 -->
    <param name="gap_y_offset" value="0"/>

    <param name="gap_left_width" value="0.03"/>
    <param name="gap_right_width" value="0.029"/>

    <param name="start_state_x" value="-1.5"/>
    <param name="start_state_y" value="1"/>
    <param name="start_state_theta" value="-1.57"/>
    <param name="start_state_joint1" value="0"/>
    <param name="start_state_joint2" value="1.57"/>
    <param name="start_state_joint3" value="0"/>

    <param name="goal_state_x" value="3"/>
    <param name="goal_state_y" value="1"/>
    <param name="goal_state_theta" value="-1.57"/>

    <param name="state_validity_check_res" value="0.005"/>
    <param name="valid_segment_count_factor" value="20"/> <!--20-->

    <!-- for transform controller -->
    <rosparam file="$(find hydrus)/config/hydrusx/$(arg model)/Hydrus.yaml" command="load" />
    <param name="debug_log" type="bool" value="false"/>

    <param name="solving_time_limit" type="double" value="3600"/>

    <param name="semi_stable_cost" type="double" value="0.5"/>
    <param name="full_stable_cost" type="double" value="0.5"/>

    <param name="length_opt_weight" type="double" value="1.0"/>
    <param name="stability_opt_weight" type="double" value="0"/>

    <param name="stability_cost_thre" type="double" value="1.2" />
    <param name="length_cost_thre" type="double" value="6" /> <!-- 6 -->

    <param name="log_flag" type="bool" value="false" />

    <!-- for motion control -->
    <param name="play_log_path" type="bool" value="false"/>
    <param name="file_name" value="$(find gap_passing)/path_log/planning_log.txt" />

    <remap from="/hydrus/joints_ctrl" to="/hydrusx/joint_states" />

  </node>
</launch>
